server/web.js

```diff
@@ -314,3 +381,427 @@ function getPluginVersion(callback) {
+  var fileLines = fileText.split("\n");
+  var regexesList = regexes.split(";;");
+
+  for (let lineNumber = 1; lineNumber < fileLines.length + 1; lineNumber++) {
+    // ';;' is the delimiter between regexes
+    regexesList.forEach(function(regex) {
+      if (regex.length > 0) {
+
+        // Keeps track of what part of the line we start at, since if we're finding
+        // multiple of the same regex in the same line, we need to cut off the first
+        //   part of the line
+        var indexInLine = 0;
+
+        while (indexInLine < fileLines[lineNumber-1].length) {
+
+          var result = child_process.spawnSync('tre-agrep',
+            ['--show-position', '--line-number', '--regexp', regex, '-'],
+            {'input': fileLines[lineNumber - 1].substring(indexInLine)}
+          );
+
+          var mapValue = getRegexLocationAndLength(result.stdout);
+          if (mapValue) {
+            // Then we got a regex match
+
+            // Its actual indexInLine from the start of the entire line
+            //   depends on where our substring started (indexInLine)
+            mapValue["indexInLine"] = indexInLine + mapValue["indexInLine"];
+            // Start the next substring from where this regex ends
+            indexInLine = mapValue["indexInLine"] + mapValue["length"];
+
+            if (regexesMap.has(lineNumber)) {
+              regexesMap.set(lineNumber, regexesMap.get(lineNumber).concat([mapValue]));
+            } else {
+              regexesMap.set(lineNumber, [mapValue]);
+            }  
+          } else {
+            // If no mapValue, there are no more regexes on this line
+            // so we're done with the while loop
+            break;
+          }
+
+
+        }
+
+        
+      }
+    });
+  }
+
+  return regexesMap;
+}
+
+/** Get the location within a line and length of a regex match,
+ * given the result of a tre-agrep call */
+function getRegexLocationAndLength(stdout) {
+  if (!stdout) {
+    return;
+  }
+
+  // stdout returns ASCII numbers, so convert them to strings
+  var resultString = stdout.toString('utf8');
+
+  var values = resultString.split(':');
+  if (values.length < 3) {
+    // Not a legitimate match
+    return;
+  }
+
+  var lineNumber = parseInt(values[0]);
+  var relevantChars = values[1];
+  var indices = relevantChars.split('-');
+  var indexInLine = parseInt(indices[0]);
+  // Note: If *, only includes the len of things before the *
+  //   haven't tested if you have abc*xyz as the regex yet
+  var lengthToHighlight = parseInt(indices[1]) - parseInt(indices[0]);
+
+  var mapValue = {
+    'indexInLine': indexInLine,
+    'length': lengthToHighlight
+  };
+
+  return mapValue;
+}
+
+
+function getChunkedDiffs(ops, threshold, baseline) {
+    if (!threshold) {
+      threshold = 10000;
+    }
+    // TODO: Very large threshold => no results
+
+    // If there have been no changes to the document,
+    // ops = {v:0}
+    if (!Array.isArray(ops)) {
```

Comment:
Really? This seems like problem with the f'n that returns the ops -- fix there.

Reply:
done.